chainer/reporter.py

```diff
@@ -0,0 +1,325 @@
+import collections
+import contextlib
+
+import numpy
+import six
+
+from chainer import cuda
+from chainer import variable
+
+
+class Reporter(object):
+
+    """Object to which observed values are reported.
+
+    Reporter is used to collect values that users want to watch. The reporter
+    object holds a mapping from value names to the actually observed values.
+
+    When a value is passed to the reporter, an observer object can be
+    optionally attached. In this case, the name of the observer is added as the
+    prefix of the value name. The observer name should be registered
+    beforehand.
+
+    See the following example::
+
+       >>> from chainer import Reporter, report, report_scope
+       >>>
+       >>> reporter = Reporter()
+       >>> observer = object()  # it can be an arbitrary (reference) object
+       >>> reporter.add_observer('my_observer:', observer)
+       >>> observation = {}
+       >>> with reporter.scope(observation):
```

Comment:
I supposed users would encounter this example before they read the document of `scope` method, as I did in reviewing. So I thought it would make easier to understand that we write the behavior of `scope`. But, since it is natural to assume that they will read the document of `scope` right after reading this example, as you point out, only brief description is enough.